v2/internal/crdmanagement/helpers.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package crdmanagement
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+
+	"github.com/Azure/azure-service-operator/v2/internal/config"
+	"github.com/Azure/azure-service-operator/v2/internal/set"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/registration"
+)
+
+func GetNonReadyCRDs(ctx context.Context, cfg config.Values, crdManager *Manager) (map[string]apiextensions.CustomResourceDefinition, error) {
+	existingCRDs, err := crdManager.ListOperatorCRDs(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list operator CRDs")
+	}
+
+	goalCRDs, err := crdManager.LoadOperatorCRDs(CRDLocation, cfg.PodNamespace)
```

Comment:
You might have some rebase conflicts from my recently merged "Don't list CRDs twice" PR

Reply:
Yep, fixed it already. Thanks!